src/base_client.ts

```diff
@@ -81,6 +83,10 @@ export default class BaseClient {
           }
         }
       });
-  });
+    });
+  }
+
+  private sanitisePath(path: string): string {
+    return path.replace(/\/ /g, "/").replace(/\/ $/, "");
```

Comment:
I wonder if we can avoid the `normalizePath` call altogether  and replace it with just the regex in `sanitisePath`. Also, we should decide on either US or UK spelling :)